zebra-network/src/meta_addr.rs

```diff
@@ -422,19 +398,84 @@ impl MetaAddr {
         self.last_failure
     }
 
-    /// Is this address a directly connected client?
-    pub fn is_direct_client(&self) -> bool {
-        match self.last_connection_state {
-            Responded => !self.services.contains(PeerServices::NODE_NETWORK),
-            NeverAttemptedGossiped | NeverAttemptedAlternate | Failed | AttemptPending => false,
+    /// Have we had any recently messages from this peer?
+    ///
+    /// Returns `true` if the peer is likely connected and responsive in the peer
+    /// set.
+    ///
+    /// [`constants::LIVE_PEER_DURATION`] represents the time interval in which
+    /// we should receive at least one message from a peer, or close the
+    /// connection. Therefore, if the last-seen timestamp is older than
+    /// [`constants::LIVE_PEER_DURATION`] ago, we know we should have
+    /// disconnected from it. Otherwise, we could potentially be connected to it.
+    pub fn was_recently_live(&self) -> bool {
+        if let Some(last_response) = self.last_response {
+            // Recent times and future times are considered live
+            last_response.saturating_elapsed()
+                <= constants::LIVE_PEER_DURATION
+                    .try_into()
+                    .expect("unexpectedly large constant")
```

Comment:
Just an idea: if `LIVE_PEER_DURATION` is `Duration32`, it can be converted to `{std::time, chrono}::Duration` when needed without `.try_into().expect(...)`. Not sure how many conversions there are, so it might not be a good idea 😅

Reply:
It's compared to both `std::time::Duration` and `Duration32`, and there are more `std::time::Duration` comparisons.

We could hide the `std::time::Duration`s using a `From` impl. I've added that to #2292.